docs/zoneinfo.rst

```diff
@@ -0,0 +1,370 @@
+system time zone data is available, the library will fall back to using the
+first-party ``tzdata`` package available on PyPI.
+
+.. seealso::
+
+    Module: :mod:`datetime`
+        Provides the time and datetime types with which the ZoneInfo class was
+        designed to be used.
+
+    Package `tzdata <https://pypi.org/project/tzdata/>`_
+        First-party package maintained by the CPython core developers to supply
+        time zone data via PyPI.
+
+
+Using ``ZoneInfo``
+------------------
+
+:class:`ZoneInfo` is a concrete implementation of the :class:`datetime.tzinfo`
+abstract base class, and is intended to be attached to ``tzinfo``, either via
+the constructor, the ``.replace`` method or ``.astimezone``::
+
+    >>> from zoneinfo import ZoneInfo
+    >>> from datetime import datetime, timedelta
+
+    >>> dt = datetime(2020, 10, 31, 12, tzinfo=ZoneInfo("America/Los_Angeles"))
+    >>> print(dt)
+    2020-10-31 12:00:00-07:00
+
+    >>> dt.tzname()
+    'PDT'
+
+Datetimes constructed thusly are compatible with datetime arithmetic and handle
+daylight saving time transitions with no further intervention::
+
+    >>> dt_add = (dt + timedelta(days=1))
+
+    >>> print(dt_add)
+    2020-11-01 12:00:00-08:00
+
+    >>> dt_add.tzname()
+    'PST'
+
+These time zones also support the ``fold`` attribute introduced in :pep:`495`.
+During offset transitions which induce ambiguous times (such as a daylight
+saving time to standard time transition), the offset from *before* the
+transition is used when ``fold=0``, and the offset *after* the transition is
+used when ``fold=1``, for example::
+
+    >>> dt = datetime(2020, 11, 1, 1, tzinfo=ZoneInfo("America/Los_Angeles"))
+    >>> print(dt)
+    2020-11-01 01:00:00-07:00
+
+    >>> print(dt.replace(fold=1))
+    2020-11-01 01:00:00-08:00
+
+When converting from another time zone, the fold will be set to the correct
+value::
+
+    >>> from datetime import timezone
+    >>> LOS_ANGELES = ZoneInfo("America/Los_Angeles")
+    >>> dt_utc = datetime(2020, 11, 1, 8, tzinfo=timezone.utc)
+
+    >>> print(dt_utc.astimezone(LOS_ANGELES))
+    2020-11-01 01:00:00-07:00
+
+    >>> print((dt_utc + timedelta(hours=1)).astimezone(LOS_ANGELES))
+    2020-11-01 01:00:00-07:00
+
+Data sources
+------------
+
+The ``zoneinfo`` module does not directly provide time zone data, and instead
+pulls time zone information from the system time zone database or the
+first-party PyPI package ``tzdata``, if available. Some systems, including
+notably Windows systems, do not have an IANA database available, and so for
+projects targeting cross-platform compatibility that require time zone data, it
+is recommended to declare a dependency on ``tzdata``.
+
+.. _zoneinfo_data_configuration:
+
+Configuring the data sources
+****************************
+
+When ``ZoneInfo(key)`` is called, the constructor first searches the
+directories specified in :data:`TZPATH` a match to ``key``, and on failure
```

Comment:
It's either a typo or I cannot parse it:
```suggestion
directories specified in :data:`TZPATH` to match a ``key``, and on failure
```

Reply:
a `+1` to `webknjaz`

( Except maybe, "to match the" )

Reply:
Right, I figured that he may have accidentally swapped the words. So I changed only that.
Patched my suggestion.

Reply:
How about "directories specified in TZPATH for a file matching `key`"?

Reply:
That reads well.